checklists/lz_checklist.en.json

```
@@ -853,6 +853,7 @@
             "text": "Consider using Bastion to securely connect to your network.",
             "guid": "ee1ac551-c4d5-46cf-b035-d0a3c50d87ad",
             "severity": "Medium",
+            "graph": "resources | summarize compliant=countif(id contains 'microsoft.network/bastionhosts') by id | where compliant > 0 | project id, compliant",
```

I think being able to project by name would be helpful - should we consider using the query as:

```suggestion
            "graph": "resources | summarize compliant=countif(id contains 'microsoft.network/bastionhosts') by name, id | where compliant > 0 | project name, id, compliant",
```

---

@brsteph I wrote the query to output the 2 fields required (id, compliant), but showing the name is probably more intuitive so looks good to me!

---

@brsteph @BeatrizSilv do not forget that these queries are not natively used by users, but through some tooling. For example, with the Azure Monitor Workbooks that automatically render an ID as a linkable resource name. Adding the name is not necessarily add a lot of value to the query.

---

My concern is rather whether this query would highlight VNets not accessible by Bastion? As far as I can see, this query would only return "compliant" if there is a Bastion, but wouldn't return "non-compliant" if there is not one.

---

> @brsteph @BeatrizSilv do not forget that these queries are not natively used by users, but through some tooling. For example, with the Azure Monitor Workbooks that automatically render an ID as a linkable resource name. Adding the name is not necessarily add a lot of value to the query.

Understood - my thought was just what a human would see in the dashboard, more than the tooling.  So if it isn't helpful for the purpose of the query, then ignore the feedback!

---

@erjosito , @brsteph I came up with this new version of the query, though still not ideal:

Resources 
|where id contains 'b63ce932-8f9f-4a24-a6f5-354b9dc7e808'
|where type == 'microsoft.network/virtualnetworks'
|where properties contains 'GatewaySubnet'
|extend compliant=properties contains 'AzureBastionSubnet'
|project id, compliant
|union (
    Resources
    |where id contains 'b63ce932-8f9f-4a24-a6f5-354b9dc7e808'
    |where type == 'microsoft.network/virtualnetworks'
    |where properties contains 'remotevnettohubpeering'
    |extend compliant=properties contains 'azurebastionsubnet'|project id, compliant)

It basically checks if every hub VNET has a Bastion in it, as well as VNETs peered to a vhub, and then combines the output of both queries in a single table. It works as expected, but not sure if it will not cause some confusion during reviews because you will have a lot of VNETs linked to vhubs that don't have a Bastion, but that are not necessarily 'not compliant'.

I tried writing the second query in a different way (check if at least of the VNets peered to a vhub have a Bastion to consider as 'compliant') but it proved to be more difficult than I thought, plus not sure if it would be compatible with the workbook output.

Let me know what you think.

---

You picked a hard one for your first query :-) Yes, it would be confusing if we show VNets peered to vhubs as non-compliant, even if there is a Bastion in another spoke. We might want to turn this PR into draft, and come back to it later...

---

It might be that we need three settings:

- Compliant (Bastion in Vnet)
- Compliant (Bastion in peered Vnet)
- Non-Compliant

But I know Widgets in the portal today run in to the same issue that you are having here!

---

@brsteph for hub and spoke yes, for VWAN it is tougher, because Bastion could be in another spoke. Not sure if it is easy, but I would try something like:

- VWAN hub: compliant if it has a connection to a VNet with an "AzureBastionSubnet"
- H&S hub (VNet including a "GatewaySubnet"): compliant if it has a peering to a VNet with an "AzureBastionSubnet"

---

Looks good to me.